### PR TITLE
Import Application Licence Migration

### PR DESCRIPTION
**Pull request by mikeylockwood**

### CODE REVIEW

File: data_migration/models/base.py [hunk 1]
```diff
@@ -103,6 +103,12 @@ def get_m2m_data(cls, target: models.Model) -> Generator:
         return cls.get_source_data()
 
 
+class CaseReference(MigrationBase):
+    prefix = models.CharField(max_length=8)
+    year = models.IntegerField(null=True)
```
Comment (MattHolmes123) [inline data_migration/models/base.py:108]: So in the v2 model we have this comment:
```
# this is null for importer/exporter access requests and mailshots
year = models.IntegerField(null=True)
```
Are we going to be migrating access requests and mailshots?
(I don't actually know what mailshots are tbh)

File: data_migration/models/import_application/import_application.py [hunk 2]
```diff
@@ -84,14 +86,34 @@ class ImportApplication(MigrationBase):
 
     imi_submit_datetime = models.DateTimeField(null=True)
 
+    # TODO ICMSLST-1694: licence_reference is FK to CaseReference
+    # TODO ICMSLST-1677: variation_requests M2M
+    # TODO ICMSLST-1678: further_information_requests M2M
+    # TODO ICMSLST-1675: update_requests M2M
+    # TODO ICMSLST-1675: case_notes M2M
+    # TODO ICMSLST-1675: case_emails M2M
```
Comment (MattHolmes123) [inline data_migration/models/import_application/import_application.py:94]: You're brave doing all this in ICMSLST-1675 :)

File: data_migration/models/base.py (hunk 1 above)
Comment (mikeylockwood) [reply data_migration/models/base.py:108]: Yes `ICMSLST-1674` and `ICMSLST-1679`

File: data_migration/models/import_application/import_application.py (hunk 2 above)
Comment (mikeylockwood) [reply data_migration/models/import_application/import_application.py:94]: i don't know what this data is yet so I have a generic ticket for response data at the moment

File: data_migration/models/import_application/import_application.py [hunk 3]
```diff
@@ -192,6 +214,37 @@ def get_excludes(cls) -> list[str]:
         return super().get_excludes() + ["imad_id"]
 
 
+class ImportCaseDocument(MigrationBase):
```
Comment (MattHolmes123) [inline data_migration/models/import_application/import_application.py:217]: Do you plan on having an explicit ExportCaseDocument class later on?

Comment (mikeylockwood) [reply data_migration/models/import_application/import_application.py:217]: Yes as each class will be using a different `content_type_id` when migrating to V2

File: data_migration/tests/test_e2e.py [hunk 4]
```diff
@@ -113,18 +129,10 @@
...
-    factory.CountryFactory(id=1000, name="My Test Country")
-    cg = dm.CountryGroup.objects.create(country_group_id="SIL", name="SIL")
-
     ocg = dm.ObsoleteCalibreGroup.objects.create(name="Test OC Group", order=1, legacy_id=1)
     dm.ObsoleteCalibre.objects.create(legacy_id=444, calibre_group=ocg, name="Test OC", order=1)
 
-    process_pk = max(web.Process.objects.count(), dm.Process.objects.count()) + 1
-    pk_range = list(range(process_pk, process_pk + 2))
-    iat = factory.ImportApplicationTypeFactory(master_country_group=cg)
+    call_command("export_from_v1")
```
Comment (MattHolmes123) [inline data_migration/tests/test_e2e.py:135]: I did not know about call_command :+1:

**Review by MattHolmes123 — APPROVED** [review]

File: data_migration/models/import_application/import_application.py (hunk 3 above)
Comment (MattHolmes123) [reply data_migration/models/import_application/import_application.py:217]: yer cool makes sense.

**Pull request merged**